pkg/webhook/preflight/nutanix/imagekubernetesversioncheck.go

```diff
@@ -92,6 +98,17 @@ func (c *imageKubernetesVersionCheck) checkKubernetesVersion(image *vmmv4.Image)
 	return nil
 }
 
+// sanitizeKubernetesVersion parses the Kubernetes version and returns major.minor.patch.
+// For example, "1.33.1+fips.0" becomes "1.33.1".
+func sanitizeKubernetesVersion(version string) (string, error) {
+	parsedVersion, err := semver.Parse(version)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%d.%d.%d", parsedVersion.Major, parsedVersion.Minor, parsedVersion.Patch), nil
```

Comment:
Can use [FinalizeVersion](https://github.com/blang/semver/blob/master/v4/semver.go#L66C18-L66C33) for this.